js/directives/day-block.js

```diff
@@ -0,0 +1,63 @@
+/* global angular */
+(function(){
+    'use strict';
+    var DayBlockDirective = function () {
+        return {
+            template : 
+            '<div class="day-bloc">'+
+            
+            '<div ng-style = \'{'+
+            'color: controller.firstColor'+
+            '}\' '+
+            'ng-if = "controller.showDay" class="weekday">{{controller.date | luifMoment: \'dddd\'}}'+
+            '</div>'+
+            
+            '<div ng-style = \'{ '+
+            'border: "1px solid " + controller.firstColor, '+
+            '"background-color": controller.firstColor, '+
+            '"color": controller.secondColor '+
+            '}\' class="day">{{controller.date | luifMoment:\'DD\'}}'+
+            '</div>'+
+            
+            '<div ng-style = \'{'+
+            '"background-color": controller.secondColor, '+
+            ' "border-left" : "1px solid " + controller.firstColor,'+
+            ' "border-right" : "1px solid " + controller.firstColor, '+
+            ' color: controller.firstColor '+
+            '}\' class="month">{{controller.date | luifMoment: \'MMM\' | limitTo : 3}}'+
+            '</div>'+
+            
+            '<div ng-style = \'{'+
+            '"background-color": controller.secondColor, '+
+            ' "border-left" : "1px solid " + controller.firstColor,'+
+            ' "border-right" : "1px solid " + controller.firstColor, '+
+            ' "border-bottom" : "1px solid " + controller.firstColor, '+
+            ' color: controller.firstColor '+
```

Comment:
I'm not a big fan of using functions like weekdayStyleOverride. I know it's more readable but as angular it doesn't know if the returned value by weekdayStyleOverride is changed. It will call it again and again draining cpu resources and as result of all these calls the garbage collectors will be called to clean all the unused object created at every weekdayStyleOverride call.